[rocprim] Successful build in clean chroot

tpkessler opened the pull request:
See #119

s-bernard commented:
Too many commits not belonging to rocprim even if the changes are OK. A simple `git rebase upstream master` should do the trick (where upstream is this repo).

tpkessler commented:
Hey @s-bernard, thanks for your review. You're right, there are too many commits unrelated to `rocprim`. I did `git rebase` before but it seems like I messed it up. I've tried it again with the result that it's not getting better. Any suggestions?

s-bernard commented:
OK, do `git rebase -i rocm-arch master` (replace rocm-arch by the name of the remote representing this repo) and remote all lines of commit you do not want.
You can reorder commits (lines) (order is top to bottom, and not bottom to top like in `git log`) or squast commits together (by putting `s` instead of pick behind a commit, it will be squash with the previous one) and reword your descriptions. No need to do all in one pass, you can relaunch the rebase -i after.

s-bernard commented:
Maybe the command is `git rebase -i rocm-arch/master` I never remember.

tpkessler commented:
Thanks! I totally forgot about the `-i` flag! It's now down to the two commits  that actually change `rocprim`.

s-bernard commented:
Last thing about the git subject, you should squash the two commits (having a separate commit for the single quotes is not really useful) and reword the description.

acxz commented on rocprim/PKGBUILD:
```diff
@@ -2,43 +2,33 @@
-        -Damd_comgr_DIR=/opt/rocm/lib/cmake/amd_comgr \
+        -DBUILD_BENCHMARK=OFF \
         "$srcdir/rocPRIM-rocm-$pkgver"
-  make
```
Shouldn't we still be `make`ing the package?

tpkessler replied:
rocprim is a header-only library. That's why you only need `make install` but not `make`

acxz replied:
I see, sounds good.

acxz commented on rocprim/PKGBUILD:
```diff
@@ -6,7 +6,7 @@ pkgdesc='Header-only library providing HIP parallel primitives'
 url="https://github.com/ROCmSoftwarePlatform/rocPRIM"
 license=('MIT')
-depends=('hip')
+depends=('hip-hcc')
```
Does rocprim work with hip-nvcc and hip-clang? Shouldn't this stay generic?

tpkessler replied:
rocprim only works on the ROCm platform. The generic version for ROCm and CUDA is hipcub.

acxz replied:
I see kk

acxz commented:
I tried building a clean chroot, but got the following error: 
```
CMake Error at /usr/share/cmake-3.17/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findamd_comgr.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "amd_comgr", but CMake did not find one.

  Could not find a package configuration file provided by "amd_comgr" with
  any of the following names:

    amd_comgrConfig.cmake
    amd_comgr-config.cmake

  Add the installation prefix of "amd_comgr" to CMAKE_PREFIX_PATH or set
  "amd_comgr_DIR" to a directory containing one of the above files.  If
  "amd_comgr" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /opt/rocm/hip/lib/cmake/hip/hip-config.cmake:78 (find_dependency)
  cmake/VerifyCompiler.cmake:58 (find_package)
  CMakeLists.txt:47 (include)
```

tpkessler commented:
Hey @acxz, thanks a lot for your review! I added the missing config line for cmake and added git as a missing build dependency. Keep in mind that I'm assuming that hip-hcc depends on comgr (see my other PR), so you may also install this package when building in a clean chroot.

acxz commented:
Sorry this took a while, but with the newer hip-hcc I got it to build in clean chroot. LGTM
